tempodb/encoding/vparquet4/block_autocomplete.go

```diff
@@ -533,6 +620,13 @@ func createDistinctSpanIterator(
 			}
 		}
 
+		// = nil ?
+		if isNotExistSearch {
+			pred := parquetquery.NewIncludeNilStringEqualPredicate([]byte(cond.Attribute.Name))
+			iters = append(iters, makeNilIter(columnPathSpanAttrKey, pred, "")) // don't select just filter nils
```

Comment:
Confirming my understanding:  this line (and the others like it) prevent the panic by not dropping into the genericConditions for =nil checks on generic attrs.  Because the Operands[0].Type lookup is still in `createDistinctAttributeCollector`, instead we are catching that one layer higher here.